test/e2e/validation_test.go

```diff
@@ -545,22 +545,27 @@ func (tc *testContext) testInvalidServicesCM(cmName string) error {
 	return nil
 }
 
-// waitForValidWindowsServicesConfigMap returns a reference to the ConfigMap that matches the given name.
-// If a ConfigMap with valid contents is not found within the time limit, an error is returned.
-func (tc *testContext) waitForValidWindowsServicesConfigMap(cmName string) (*core.ConfigMap, error) {
+// waitForValidWindowsServicesConfigMap waits for a ConfigMap by the given name to either be created or deleted,
+// depending on the value of the `deletion` flag. If the flag is false, it tries to retrieve and return a ConfigMap with
+// valid contents. Returns an error if it cannot be not found by the time limit. If the flag is true, it ensures that
+// the ConfigMap cannot be found within the WMCO namespace. Returns an error if it is still present at the time limit.
+func (tc *testContext) waitForValidWindowsServicesConfigMap(cmName string, deletion bool) (*core.ConfigMap, error) {
 	configMap := &core.ConfigMap{}
 	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
```

Comment:
Is there a reason to wait `retry.Interval` and then get the ConfigMap? Why not `wait.PollImmediate()`?

Reply:
No reason, we can use `wait.PollImmediate()`